tests/test_observatory.py

```diff
@@ -247,14 +254,67 @@ def test_observatory_override(sandbox, overwrite):
     else:
         load_observatories(io.StringIO(wronggbt), overwrite=overwrite)
         newgbt = get_observatory("gbt")
-        assert newgbt._loc_itrf.y > 0
-        assert newgbt._loc_itrf.y != gbt_orig._loc_itrf.y
+        assert newgbt.y > 0
+        assert newgbt.y != gbt_orig.y
 
 
 def test_list_last_correction_mjds_runs():
     pint.observatory.list_last_correction_mjds()
 
 
+def test_TopoObs_EarthLocation(sandbox):
+    gbt_orig = get_observatory("gbt")
+    gbt_new = TopoObs(name="gbt_new", location=gbt_orig.location)
+    assert gbt_orig.separation(gbt_new) < 1 * u.cm
+
+
+def test_json_observatory_output(sandbox):
+    gbt_orig = get_observatory("gbt")
+    load_observatories(io.StringIO(gbt_orig.as_json), overwrite=True)
+    gbt_reload = get_observatory("gbt")
+
+    for p in gbt_orig.__dict__:
+        if not p in ["_clock"]:
+            assert getattr(gbt_orig, p) == getattr(gbt_reload, p)
+
+
+def test_json_observatory_output_latlon(sandbox):
+    gbt_orig = get_observatory("gbt")
+    gbt_dict = gbt_orig.as_dict
+    # remove ITRF
+    del gbt_dict["gbt"]["itrf_xyz"]
+    # add in geodetic
+    gbt_dict["gbt"]["lat"] = gbt_orig.lat.value
+    gbt_dict["gbt"]["lon"] = gbt_orig.lon.value
+    gbt_dict["gbt"]["alt"] = gbt_orig.alt.value
+    load_observatories(io.StringIO(json.dumps(gbt_dict)), overwrite=True)
+    gbt_reload = get_observatory("gbt")
+
+    for p in gbt_orig.__dict__:
+        if not p in ["location", "_clock"]:
+            # everything else should be identical
+            assert getattr(gbt_orig, p) == getattr(gbt_reload, p)
+    # check distance separately to allow for precision
+    distance = gbt_orig.separation(gbt_reload)
+    assert distance < 1 * u.m
+
+
+def test_json_observatory_output_latlon_and_itrf(sandbox):
```

Comment:
Am I right in understanding that this checks that an error is emitted if you have both xyz and lat/lon/alt or if you're missing lat but doesn't check that lat/lon/alt works? If so can the test name make that clear? Is specifying lat/lon/alt tested elsewhere?

Reply:
I changed this name to `test_json_observatory_input_latlon_and_itrf_giveserror`
And yes, `test_json_observatory_input_latlon` checks that inputing an observatory via lat/lon works (and gives the same info)